File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int  zeros = 0, l = 0, r = 0;
        for(r = 0; r <nums.size(); r++){
            if(nums[r]==0) zeros++;
            //if zeros exceed target then we increment l and check whether it's zero or not , if 'yes' then zeros--
            if(zeros > k){
                if(nums[l] == 0) zeros--;
                l++;
            }
        }
        return r - l ;   
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] arr, int k) {
        int l=0,r=0,zeros=0;
        int maxLen = Integer.MIN_VALUE;
        while(r<arr.length){
            if(arr[r]==0) zeros++;
            if(zeros > k){
                if(arr[l]==0)zeros--;
                l++;
            }
            if(zeros <= k) maxLen = Math.max(maxLen,r-l+1);
            r++;
        }
        return maxLen;
    }
}